chcemvediet/apps/inforequests/admin.py

```diff
@@ -315,3 +317,58 @@ def get_queryset(self, request):
         queryset = queryset.select_related(u'branch')
         queryset = queryset.select_related(u'email')
         return queryset
+
+    def get_actions(self, request):
+        actions = super(ActionAdmin, self).get_actions(request)
+        if u'delete_selected' in actions:
+            del actions[u'delete_selected']
+        return actions
+
+    def has_delete_permission(self, request, obj=None):
+        if obj is None:
+            return True
+        if obj.type in [Action.TYPES.REQUEST, Action.TYPES.ADVANCED_REQUEST]:
+            return False
+        if len(obj.branch.actions) > 1:
+            return True
+        return False
+
+    def render_delete_form(self, request, context):
+        using = router.db_for_write(self.model)
+        collector = admin.utils.NestedObjects(using)
+        collector.collect([context[u'object']])
+        to_delete = collector.nested()
+        for object in traverse(to_delete):
+            if isinstance(object, Action) and object.email:
+                for inforequestemail in object.email.inforequestemail_set.filter(
+                        type=InforequestEmail.TYPES.APPLICANT_ACTION):
```

Comment:
@martinmacko47 Mam vypisat na obrazovku aj tie inforequesemaily, ktorym sa len zmeni type na `unrelated`.

Reply:
@viliambalaz Ano vypis oba. V dvoch zoznamoch:

- Action: ...
   - Branch: ...
   - Branch: ...
- Outbound messages will be deleted:
    - Inforequest email: ...
    - Inforequest email: ...
- Inbound messages will be marked undecided:
    - Inforequest email: ...
    - Inforequest email: ...

Reply:
:ok: